Add workflow for auto-updating non-default branches

tinywrkb opened the pull request:
Inspired from a similar workflow for org.winehq.Wine that was added by @gasinvein.

This runs flatpak-external-data-checker every 12 hours, and opens new PRs if something has changed.  
Like the flathubbot initiated f-e-d-c runs, if there's already an opened PR with similar changes, then another one won't be created.

The workflow can also be triggered manually from GitHub's Actions tab.  

~~This handles non-default branches, not the default `branch/5.15-21.08` which flathubbot already taking care of.  
Currently, only `branch/6.2` is set in the strategy matrix.~~  
**Update: Changed to run on all the maintained branches, and flathubbot f-e-d-c is disabled**

~~What missing here is allowing to trigger the workflow manually with `workflow_dispatch` on the default branch, and without having to add a separate workflow.  
I'll leave that for later.~~

tinywrkb committed:
Add workflow for auto-updating non-default branches

Inspired from a similar workflow for org.winehq.Wine that was added by @gasinvein.

gasinvein commented:
> This handles non-default branches, not the default `branch/5.15-21.08` which flathubbot already taking care of.

I think it would be cleaner to only f-e-d-c workflow for all branches. You can disable the "hosted" f-e-d-c by adding `"disable-external-data-checker": true` to `flathub.json`.

tinywrkb commented:
> > This handles non-default branches, not the default `branch/5.15-21.08` which flathubbot already taking care of.
> 
> I think it would be cleaner to only f-e-d-c workflow for all branches. You can disable the "hosted" f-e-d-c by adding `"disable-external-data-checker": true` to `flathub.json`.

Right! I thought about it, but I wasn't sure if it's acceptable to have higher chances to trigger multiple builds at the same time.  
We're talking about over an hour for each build.

tinywrkb commented:
Switched to draft. Not ready for merging, and I want #45 merged first.

tinywrkb committed:
Use update workflow with the default branch

Disable flathubbot initiated f-e-d-c checks, and add the default 5.15-21.08 branch
to the strategy matrix array of the update workflow.
